os.c

```diff
@@ -138,40 +134,30 @@ isc_result_t
 perf_os_waituntilanyreadable(int *fds, unsigned int nfds, int pipe_fd,
 			     isc_int64_t timeout)
 {
-	fd_set read_fds;
+	struct pollfd read_fds[nfds];
```

Comment:
How do you build this code (nfds here is not constant)?
Would be better to use
`perf_os_waituntilanyreadable(struct pollfd *read_fds, unsigned int nfds, isc_int64_t timeout)`
instead. In the code we have chain `tinfo->socks -> fds -> read_fds` and we can avoid using intermediate fds array here.
Also in dnsperf.c there are no check perf_os_waituntilanyreadable ret value - so this check also can be removed from the function.

Reply:
struct pollfd read_fds[nfds]; requires C99. Perhaps configure.ac should check for this.

The implementation of perf_os_waituntilanyreadable() uses select internally, and so has the same limitation on number of sockets. Possibly reimplementing that to use poll() and then using that makes sense.

Reply:
In configure.ac, s/AC_PROG_CC/AC_PROG_CC_C99/ for first.